### PR TITLE
IGListBindingSectionController no longer asserts when reloading the entire section. 

### PR DESCRIPTION
**Pull request by jeffbailey**
## Changes in this pull request

Issue fixed: #1174

### Checklist

- [X] All tests pass. Demo project builds and runs.
- [x] I added tests, an experiment, or detailed why my change isn't tested.
- [X] I added an entry to the `CHANGELOG.md` for any breaking changes, enhancements, or bug fixes.
- [X] I have reviewed the [contributing guide](https://github.com/Instagram/IGListKit/blob/master/.github/CONTRIBUTING.md)

### CODE REVIEW

**Comment by jeffbailey** [comment]
To test the change, I updated IGTestDiffingObject::isEqualToDiffableObject to not always return YES. Now it does a simple compare for the number of objects associated with the key. That actually triggered the assert that was there previously when running IGListBindingSectionControllerTests::test_whenAdapterReloadsObjects_thatSectionUpdated. Now that I converted the assert to a warning, the test passes again.

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,8 @@ The changelog for `IGListKit`. Also see the [releases](https://github.com/instag
 
 - Added `IGListCollectionScrollingTraits` for exposing `UICollectionView` scrolling traits to section controllers via `IGListCollectionContext`. [Adam Stern](https://github.com/adamastern) (tbd)
 
+- Support reloading a `IGListBindingSectionController`. [Jeff Bailey](https://github.com/jeffbailey) (tbd)
```
Comment (rnystrom) [inline CHANGELOG.md]: 2 nits:

- Replace tbd with the number and link to this PR (1213)
- Let's make the line a little more descriptive. Something like "no longer asserts when reloading..." etc etc.

File: Tests/Objects/IGTestDiffingObject.m [hunk 2]
```diff
@@ -28,8 +28,18 @@ - (NSString *)description {
...
+        return YES;
+    }
+    if ([object isKindOfClass:[IGTestDiffingObject class]]) {
+        /* A simple equality test that only looks at the number of objects for the key.
+           It does not currently test the equality of each of the objects. */
+        IGTestDiffingObject *testDiffingObject = (IGTestDiffingObject *)object;
+        return self.objects.count == testDiffingObject.objects.count;
+    }
+    
+    return NO;
```
Comment (rnystrom) [inline Tests/Objects/IGTestDiffingObject.m]: Noting to myself that this actually tests not tripping the assert anymore. [See comment](https://github.com/Instagram/IGListKit/issues/1174#issuecomment-403966485)

Comment (jeffbailey) [reply Tests/Objects/IGTestDiffingObject.m]: Good idea. Pull request has been updated @rnystrom.

**Comment by jeffbailey** [comment]
Yeah, I'm not sure why the stacked controller change showed up there. Just squash to a single commit.

**Comment by shial4** [comment]
When will be this released?

**Comment by lorixx** [comment]
@shial4 We made 4.0.0 release yesterday, enjoy!

**Pull request closed without merging**